lib/exchange.js

```diff
@@ -19,23 +19,25 @@ function fetchExchange (settings, cryptoCode) {
     })
 }
 
-function buy (settings, cryptoAtoms, fiatCode, cryptoCode) {
+function buy (settings, tradeEntry, tradeId) {
+  const { cryptoAtoms, fiatCode, cryptoCode } = tradeEntry
   return fetchExchange(settings, cryptoCode)
     .then(r => {
       if (r.exchangeName === 'mock-exchange') {
         return mockExchange.buy(cryptoAtoms, fiatCode, cryptoCode)
       }
-      return ccxt.trade('buy', r.account, cryptoAtoms, fiatCode, cryptoCode, r.exchangeName)
+      return ccxt.trade('buy', r.account, cryptoAtoms, fiatCode, cryptoCode, r.exchangeName, tradeId)
```

Comment:
How about passing ``tradeEntry`` to ``ccxt.trade``, and including ``tradeId`` in the ``tradeEntry`` object?